tests/components/intelliclima/test_config_flow.py

```diff
@@ -0,0 +1,133 @@
+"""Test the IntelliClima config flow."""
+
+from unittest.mock import AsyncMock
+
+from pyintelliclima.api import (
+    IntelliClimaAPIError,
+    IntelliClimaAuthError,
+    IntelliClimaDevices,
+)
+import pytest
+
+from homeassistant import config_entries
+from homeassistant.components.intelliclima.const import DOMAIN
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+DATA_CONFIG = {
+    CONF_USERNAME: "SuperUser",
+    CONF_PASSWORD: "hunter2",
+}
+
+
+async def test_form(
+    hass: HomeAssistant, mock_setup_entry: AsyncMock, mock_cloud_interface
+) -> None:
+    """Test we get the form."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] is FlowResultType.FORM
+    assert result["errors"] == {}
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        DATA_CONFIG,
+    )
+    await hass.async_block_till_done()
+
+    assert result["type"] is FlowResultType.CREATE_ENTRY
+    assert result["title"] == "IntelliClima (SuperUser)"
+    assert result["data"] == DATA_CONFIG
+
+
+@pytest.mark.parametrize(
+    ("side_effect", "errors", "patch_devices"),
+    [
+        # invalid_auth
+        (IntelliClimaAuthError, {"base": "invalid_auth"}, False),
+        # cannot_connect
+        (IntelliClimaAPIError, {"base": "cannot_connect"}, False),
+        # unknown
+        (RuntimeError("Unexpected error"), {"base": "unknown"}, False),
+    ],
+)
+async def test_form_auth_errors(
+    hass: HomeAssistant,
+    mock_setup_entry: AsyncMock,
+    mock_cloud_interface,
+    side_effect,
+    errors,
+    patch_devices,
+) -> None:
+    """Test we handle authentication-related errors and recover."""
+    mock_cloud_interface.authenticate.side_effect = side_effect
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        DATA_CONFIG,
+    )
+
+    assert result["type"] is FlowResultType.FORM
+    assert result["errors"] == errors
+
+    # Recover: clear side effect and complete flow successfully
+    mock_cloud_interface.authenticate.side_effect = None
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        DATA_CONFIG,
+    )
+
+    await hass.async_block_till_done()
+    assert result["type"] is FlowResultType.CREATE_ENTRY
+    assert result["title"] == "IntelliClima (SuperUser)"
+    assert result["data"] == DATA_CONFIG
+
+
+async def test_form_no_devices(
```

Comment:
Can you add a test that will create the configuration for the same account a second time?

> Even though the code used to check the uniqueness of a config entry is most likely touched by the happy flow tests, make sure to also test that the flow doesn't allow adding more than one unique configuration entry to reach complete coverage.

https://developers.home-assistant.io/docs/core/integration-quality-scale/rules/config-flow-test-coverage